django/db/models/fields/json.py

```diff
@@ -0,0 +1,111 @@
+import json
+
+from django import forms
+from django.core import exceptions
+from django.db.utils import NotSupportedError
+from django.utils.translation import gettext_lazy as _
+
+from . import Field
+from .mixins import CheckFieldDefaultMixin
+
+
+def convert_jsonfield_value(value, expression, connection):
+    if connection.vendor == 'oracle':
+        return value.read()
+    return value
+
+
+class JSONField(CheckFieldDefaultMixin, Field):
+    description = _('A JSON object')
+    default_error_messages = {
+        'invalid': _("Value must be valid JSON."),
+    }
+    _default_hint = ('dict', '{}')
+
+    def __init__(self, encoder=None, decoder=None, default=dict, *args, **kwargs):
+        self.encoder, self.decoder = encoder, decoder
+        super().__init__(default=default, *args, **kwargs)
+
+    def _check_json_support(self, connection):
+        if not connection.features.supports_json:
+            raise NotSupportedError(_('This database backend does not support JSONField.'))
+
+    def db_check(self, connection):
+        data = self.db_type_parameters(connection)
+        if connection.vendor in ('mysql', 'sqlite'):
+            return 'JSON_VALID(%(qn_column)s)' % data
```

Comment:
We have a native `json` field on MySQL, so is this necessary?

Reply:
On MySQL, it's not, as automatic validation is provided with the [JSON data type](https://dev.mysql.com/doc/refman/5.7/en/json.html).
However, on MariaDB [prior to 10.4.3](https://mariadb.com/kb/en/library/json_valid/#description), it's necessary, because MariaDB stores JSON data with [`LONGTEXT` data type](https://mariadb.com/kb/en/library/json-data-type/).

Reply:
We support also MariaDB 10.1, so it seems that in `MySQL` backend we should change `supports_json` flag for MariaDB < 10.2.7.

Reply:
Yeah, but I think I already did so... :smile: 
https://github.com/django/django/pull/11452/files#diff-b683c76592fdc70b1d15f0e4695d12a6R125

Reply:
Great I missed that.